plugins/workspace-plugin/src/workspace-folder-updater.ts

```diff
@@ -0,0 +1,91 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as theia from '@theia/plugin';
+const UPDATE_WORKSPACE_FOLDER_TIMEOUT = 5000;
+
+export class WorkspaceFolderUpdater {
```

Comment:
All this code just because VS Code can't be bothered to return a Promise from updateWorkspaceFolders. Grr...

Reply:
it's because `updateWorkspaceFolders()` is [too tricky](https://github.com/eclipse-theia/theia/blob/4e1c1d429226280e6f0eff49658c733a38b50c4a/packages/plugin/src/theia.d.ts#L5428). I played a lot with it to avoid bugs within working on the issue...

Reply:
Yes, just ranting ;-)